detection_rules/rule.py

```diff
@@ -700,6 +700,13 @@ class QueryRuleData(BaseRuleData):
     language: definitions.FilterLanguages
     alert_suppression: Optional[AlertSuppressionMapping] = field(metadata=dict(metadata=dict(min_compat="8.8")))
 
+    @cached_property
+    def index_or_dataview(self) -> list[str]:
+        """Return the index or dataview depending on which is set."""
+        if self.index is not None:
```

Comment:
ES|QL rules have no index or dataview as those are defined in the query itself, so we may have instances where both of these conditions are not met.